lambeq/backend/quantum.py

```diff
@@ -1162,3 +1164,482 @@ def CRz(phi, distance=1): return Controlled(Rz(phi), distance)  # noqa: E731
     'CRy': CRy,
     'CRz': CRz,
 }
+
+
+def is_circuital(diagram: Diagram) -> bool:
+    """
+    Takes a :py:class:`lambeq.quantum.Diagram`,
+    checks if a diagram is a quantum 'circuital' diagram.
+    A circuital diagram is a diagram with qubits at the top.
+
+    Used to check for measurements and post-selection
```

Comment:
Is this referring to what the case was in the previous version? Is this information useful at all here?